packages/eslint-plugin/tests/rules/no-unnecessary-type-assertion.test.ts

```diff
@@ -1138,6 +1138,40 @@ var x = 1;
     },
     {
       code: `
+type S = 10;
+
+class T {
+  readonly a = 'a' as const;
+  readonly b = 3 as 3;
+  readonly c = 10 as S;
+}
```

Comment:
[Testing] One thing under test per case please (https://typescript-eslint.io/contributing/pull-requests#granular-unit-tests).

I think it would be useful having valid cases, like for a non literal class property, or a class property without the `readonly` modifier.

Reply:
I should have noticed it from the first comment 🥲
Thank you for pointing it out! I’ve split it in d000227!